tests/unit/models/test_projects.py

```diff
@@ -2125,6 +2172,40 @@ def test_build_planner_get_build_plan(platforms, expected_build_infos):
             ],
             id="no_arch",
         ),
+        pytest.param(
+            [
+                {
+                    "build-on": ["s390x"],
+                    "build-for": ["all"],
+                },
+            ],
+            [
+                BuildInfo(
+                    build_on="s390x",
+                    build_for="all",
+                    base=BaseName(name="ubuntu", version="22.04"),
+                    platform="all",
+                )
+            ],
+            id="all",
+        ),
+        pytest.param(
+            [
+                {
+                    "build-on": "s390x",
+                    "build-for": "all",
+                },
+            ],
+            [
+                BuildInfo(
+                    build_on="s390x",
+                    build_for="all",
+                    base=BaseName(name="ubuntu", version="22.04"),
+                    platform="all",
+                )
+            ],
+            id="all_as_string",
```

Comment:
What about

```
platforms:
  all:
```

I think this is invalid, but we should check.

Reply:
I agree.
```yaml
platforms:
  all:
```

On the other hand, this is probably valid:
```yaml
platforms:
  all:
    build-on: [amd64, s390x]
```

But it raises an error because I didn't update the validator to accept that.  I would like to keep this restricted for now because we haven't agreed on it as a team.  We can always support this in the future.

I added unit tests to reflect both scenarios.